From d4ba91ebe7de6c30ba3e82c86265610d4c467949 Mon Sep 17 00:00:00 2001
From: Enrico Butera <ebutera@users.berlios.de>
Date: Wed, 18 Jan 2012 15:05:10 +0100
Subject: [PATCH 13/13] board-igep0020: force twl4030 codec registers reset at
 boot

If registers are not reset to driver default values the codec will have
an unknown state.

In the igep specific case the issue is that if the codec voice interface
is disabled (REG_VOICE_IF = 0x04 in sound/soc/codec/twl4030.c) it will
never be set to that value and the expansion board UART2 will be
unusable.

Tested on IGEPv2 with expansion board rev.B

Signed-off-by: Enrico Butera <ebutera@users.berlios.de>
---
 arch/arm/mach-omap2/board-igep0020.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-igep0020.c b/arch/arm/mach-omap2/board-igep0020.c
index 3453a36..3d40f9f 100644
--- a/arch/arm/mach-omap2/board-igep0020.c
+++ b/arch/arm/mach-omap2/board-igep0020.c
@@ -684,6 +684,8 @@ static void __init igep_i2c_init(void)
 		igep_twldata.vpll2->constraints.name = "VDVI";
 	}
 
+	igep_twldata.audio->codec->reset_registers = 1;
+
 	omap3_pmic_init("twl4030", &igep_twldata);
 }
 
-- 
1.7.5.4

